spec/services/idv/agent_spec.rb

```diff
@@ -53,88 +45,71 @@
+        it 'does proof state_id if resolution succeeds' do
+          agent = Idv::Agent.new({ ssn: '444-55-8888', first_name: Faker::Name.first_name,
+                                   zipcode: '11111' })
+          result = agent.proof_resolution(proof_state_id: true)
+          expect(result[:context][:stages]).to include({ state_id: 'StateIdMock' })
         end
       end
 
-      context 'when the fist stage fails' do
-        let(:stages) { %i[failed state_id] }
-
-        it 'only the results from the first stage are included' do
-          expect(subject.to_h).to include(
-            errors: { bad: ['stuff'] },
-            messages: [failed_message],
-            success: false,
-            exception: nil,
-            timed_out: false,
-          )
+      context 'proofing state_id disabled' do
+        it 'does not proof state_id if resolution fails' do
+          agent = Idv::Agent.new({ ssn: '444-55-6666', first_name: Faker::Name.first_name,
+                                   zipcode: '11111' })
+          result = agent.proof_resolution(proof_state_id: false)
+          expect(result[:errors][:ssn]).to eq ['Unverified SSN.']
+          expect(result[:context][:stages]).to_not include({ state_id: 'StateIdMock' })
         end
-      end
-
-      context 'when the first stage times out' do
-        let(:stages) { %i[timed_out state_id] }
 
-        it 'returns a result where timed out is true' do
-          expect(subject.to_h).to include(
-            success: false,
-            timed_out: true,
-          )
+        it 'does not proof state_id if resolution succeeds' do
+          agent = Idv::Agent.new({ ssn: '444-55-8888', first_name: Faker::Name.first_name,
+                                   zipcode: '11111' })
+          result = agent.proof_resolution(proof_state_id: false)
+          expect(result[:context][:stages]).to_not include({ state_id: 'StateIdMock' })
         end
       end
 
-      context 'when there is an exception' do
-        let(:stages) { %i[exception state_id] }
+      it 'returns an unsuccessful result and notifies exception trackers if an exception occurs' do
+        exception = Proofer::TimeoutError
 
-        it 'returns an unsuccessful result and notifies exception trackers' do
-          expect(NewRelic::Agent).to receive(:notice_error).with(exception)
-          expect(ExceptionNotifier).to receive(:notify_exception).with(exception)
+        agent = Idv::Agent.new({ ssn: '444-55-8888', first_name: 'Time Exception',
+                                 zipcode: '11111' })
 
-          expect(subject.to_h).to include(
-            success: false,
-            exception: exception,
-          )
-        end
+        expect(NewRelic::Agent).to receive(:notice_error).with(exception)
+        expect(ExceptionNotifier).to receive(:notify_exception).with(exception)
+
+        expect(agent.proof_resolution(proof_state_id: false)).to include(
+          success: false,
+          exception: exception,
```

Comment:
`exception` should be an instance here, not a class, right?

Reply:
It seems like rspec figures out that it should be an instance of it, but I've split the assertions to be explicit about expecting an instance of the exception.